src/components/NetworkAssetSelectInput/NetworkAssetSelectInput.tsx

```diff
@@ -305,11 +314,52 @@ const NetworkAssetSelectInput = ({
     walletAddress,
   ]);
 
+  useEffect(() => {
+    const handleBalanceGet = async () => {
+      if (!sdk || !walletAddress) return;
+      await getSmartWalletBalancesPerChain(walletAddress, supportedChains);
+    };
+    handleBalanceGet();
+  }, [supportedChains, sdk, walletAddress]);
+
   const filteredSelectedNetworkAssets = useMemo(() => {
     const filtered = selectedNetworkAssets.filter((asset) => containsText(asset.name, assetSearchQuery) || containsText(asset.symbol, assetSearchQuery));
     return orderBy(filtered, [(asset) => asset.balanceWorthUsd ?? 0, 'name'], ['desc', 'asc']);
   }, [selectedNetworkAssets, assetSearchQuery]);
 
+  useEffect(() => {
+    const updateAvalancheBalance = () => {
+      if (
+        !sdk ||
+        !walletAddress ||
+        !preselectedNetwork ||
+        preselectedNetwork.chainId !== CHAIN_ID.AVALANCHE ||
+        !balancePerChainSmartWallet ||
+        filteredSelectedNetworkAssets.length
+      )
+        return;
+      setBalancePerChainSmartWallet([
+        ...balancePerChainSmartWallet,
```

Comment:
can be just this without pulling `balancePerChainSmartWallet` into effect:
```suggestion
      setBalancePerChainSmartWallet((current) => [
        ...current,
```